Refactored page navigation. Refactored routes protection. Resolved all bugs.

catalindanis opened the pull request:
Finished login and register bugs (page navigation, routes protection, login persistence, etc)

catalindanis committed:
Refactored page navigation. Refactored routes protection. Resolved all bugs.

Pasca22 commented on app/Reusable.tsx:
```diff
@@ -0,0 +1,26 @@
+import {ActivityIndicator, StyleSheet} from "react-native";
+import React from "react";
+
+const LoadingAnimation = () => {
```
Schimba numele fisierului sa fie `LoadingAnimation.tsx`

Pasca22 commented on app/_layout.tsx:
```diff
@@ -1,52 +1,67 @@
+
+}
+
+function NotAuthenticatedTabBar(){
```
Schimba in arrow function ca sa pastram consistenta in toata aplicatia

Pasca22 commented on app/index.tsx:
```diff
@@ -1,41 +1,30 @@
+    if (isAuthenticated)
+        return (
+            <AuthContext.Provider value={isAuthenticated}>
+                <Profile></Profile>
```
Aici ai putea folosi un singur tag (ex: `<Profile />`

Pasca22 commented on app/index.tsx:
```diff
@@ -1,41 +1,30 @@
-            </SafeAreaView>
-        </SafeAreaProvider>)
+        <AuthContext.Provider value={isAuthenticated}>
+            <Login></Login>
```
Aici ai putea folosi un singur tag (ex: <Login />

Pasca22 commented on app/Reusable.tsx:
```diff
@@ -0,0 +1,26 @@
+        </SafeAreaProvider>)
+};
+
+export default LoadingAnimation;
```
In plus, muta aceasta componenta in fisierul `Components`

Pasca22 commented on app/events.tsx:
```diff
@@ -1,14 +1,19 @@
-  
-   async function fetchEvents(): Promise<Event[]> {
+
+  async function fetchEvents(): Promise<Event[]> {
```
La fel aici, arrow function

Pasca22 commented on app/login.tsx:
```diff
@@ -1,26 +1,17 @@
-    }, [router])
+    const isAuthenticated = useContext(AuthContext);
+    if(isAuthenticated)
+        return <Profile></Profile>
```
Tot un singur tag si aici

Pasca22 commented on app/login.tsx:
```diff
@@ -54,14 +45,7 @@ const Login = () => {
             });
```
In catch-ul de mai sus, scoate acele constante daca nu le folosesti

Pasca22 commented on app/register.tsx:
```diff
@@ -1,26 +1,17 @@
-            checkAuth();
-        }, 250)
-    }, [router])
+    //const isAuthenticated = useContext(AuthContext);
+    //if(isAuthenticated)
+        //return <Profile></Profile>
```
Dc este comentat acest cod?

Pasca22 commented on app/register.tsx:
```diff
@@ -32,7 +23,7 @@ const Register = () => {
             .then((userCredential) => {
                 // Signed in
                 const user = userCredential.user;
```
Scoate toate const pe care nu le folosesti

Pasca22 commented on firebaseConfig.js:
```diff
@@ -1,6 +1,7 @@
 import { initializeApp } from "firebase/app";
 import { getAuth } from "firebase/auth";
 import { getFirestore } from "firebase/firestore";
+import {useState} from "react";
```
Dc este useState aici, daca nu e folosit?

catalindanis committed:
Resolved all comments

catalindanis committed:
modified "Components" folder name to "components"

catalindanis committed:
Added register functionality:
When a user registers, in firestore is stored it's authentication ref (uid) and other relevant fields for a user in users collection
